File: durocon/durocon/report/territory_wise_sales/territory_wise_sales.py
import frappe

def execute(filters=None):
    columns = [
        {"label": "Item Code", "fieldname": "item_code", "fieldtype": "Link", "options": "Item", "width": 120},
        {"label": "Item Name", "fieldname": "item_name", "fieldtype": "Data", "width": 200},
        {"label": "Territory", "fieldname": "territory", "fieldtype": "Link", "options": "Territory", "width": 150},
        {"label": "Total Qty", "fieldname": "total_qty", "fieldtype": "Float", "width": 120}
    ]
    
    data = frappe.db.sql("""
        SELECT 
            soi.item_code,
            soi.item_name,
            so.territory,
            SUM(soi.qty) as total_qty
        FROM `tabSales Order` so
        JOIN `tabSales Order Item` soi ON so.name = soi.parent
        WHERE so.docstatus = 1
        {conditions}
        GROUP BY so.territory, soi.item_code
        ORDER BY so.territory, soi.item_code
    """.format(conditions=get_conditions(filters)), filters, as_dict=True)
    
    return columns, data

def get_conditions(filters):
    conditions = ""
    if filters.get("from_date") and filters.get("to_date"):
        conditions += " AND so.transaction_date BETWEEN %(from_date)s AND %(to_date)s"
    if filters.get("territory"):
        conditions += " AND so.territory = %(territory)s"
    if filters.get("item_code"):
        conditions += " AND soi.item_code = %(item_code)s"
    return conditions

File: durocon/durocon/report/territory_wise_sales/territory_wise_sales.js
// // Copyright (c) 2026, Sanpra Software and contributors
// // For license information, please see license.txt

// frappe.query_reports["Territory Wise Sales"] = {
// 	"filters": [

// 	]
// }; 



frappe.query_reports["Territory Wise Sales"] = {
    "filters": [
        {
            "fieldname": "from_date",
            "label": __("From Date"),
            "fieldtype": "Date",
            "default": frappe.datetime.add_months(frappe.datetime.get_today(), -1)
        },
        {
            "fieldname": "to_date",
            "label": __("To Date"),
            "fieldtype": "Date",
            "default": frappe.datetime.get_today()
        },
        {
            "fieldname": "territory",
            "label": __("Territory"),
            "fieldtype": "Link",
            "options": "Territory"
        },
        {
            "fieldname": "item_code",
            "label": __("Item Code"),
            "fieldtype": "Link",
            "options": "Item"
        }
    ]
}